src/practices/LanguageIndependent/ReadmeCorrectlySetPractice.ts

```diff
@@ -0,0 +1,97 @@
+import { PracticeEvaluationResult, PracticeImpact } from '../../model';
+import { DxPractice } from '../DxPracticeDecorator';
+import { PracticeContext } from '../../contexts/practice/PracticeContext';
+import { PracticeBase } from '../PracticeBase';
+import { ReportDetailType } from '../../reporters/ReporterData';
+import { FixerContext } from '../../contexts/fixer/FixerContext';
+
+@DxPractice({
+  id: 'LanguageIndependent.ReadmeCorrectlySet',
+  name: 'Set README.md Correctly',
+  impact: PracticeImpact.high,
+  suggestion: 'Provide necessary sections in README',
+  reportOnlyOnce: false,
+  url: 'https://github.com/DXHeroes/knowledge-base-content/blob/master/practices/readme.md',
+  dependsOn: { practicing: ['LanguageIndependent.ReadmeIsPresent'] },
+})
+export class ReadmeIsCorrectlySet extends PracticeBase {
+  private parsedReadme: string[] = [];
+  private readmePath = 'Readme.md';
+
+  async isApplicable(): Promise<boolean> {
+    return true;
+  }
+
+  async evaluate(ctx: PracticeContext): Promise<PracticeEvaluationResult> {
+    if (!ctx.root.fileInspector) {
+      return PracticeEvaluationResult.unknown;
+    }
+
+    const parseReadme = (readmeFile: string) => {
+      return readmeFile
+        .toString()
+        .split(/\r?\n/)
+        .filter((content) => content.trim() !== '');
+    };
+
+    const regexReadme = new RegExp('readme', 'i');
+    const rootFiles = await ctx.root.fileInspector.scanFor(regexReadme, '/', { shallow: true });
+    const filePath = rootFiles[0].path;
+    this.readmePath = filePath;
+    const content = await ctx.root.fileInspector.readFile(filePath);
+
+    const parsedReadme = parseReadme(content);
+    this.parsedReadme = parsedReadme;
+
+    const heading = parsedReadme.filter((value: string) => /^(?:<h1>|#\s+)/.test(value)).length;
+    const gettingStarted = parsedReadme.find((value: string) => /^(?:<h2>|##\s+).*(Getting\s+Started)/i.test(value));
+    const prerequisites = parsedReadme.find((value: string) => /^(?:<h3>|###\s+).*(Prerequisites)/i.test(value));
+    const install = parsedReadme.find((value: string) => /^(?:<h3>|###\s+).*(Install)(?:ing|ation)/i.test(value));
+    const contribute = parsedReadme.find((value: string) => /^(?:<h2>|##\s+).*(Contribut)(?:ing|ion)/i.test(value));
+    const license = parsedReadme.find((value: string) => /^(?:<h2>|##\s+).*(License)/i.test(value));
+
+    if (heading === 1 && gettingStarted && prerequisites && install && contribute && license) {
+      return PracticeEvaluationResult.practicing;
+    }
+
+    this.setData();
+    return PracticeEvaluationResult.notPracticing;
+  }
+
+  async fix(ctx: FixerContext) {
+    const inspector = ctx.fileInspector?.basePath ? ctx.fileInspector : ctx.root.fileInspector;
+    if (!inspector) return;
+
+    const heading = this.parsedReadme.filter((value: string) => /^(?:<h1>|#\s+)/.test(value)).length;
```

Comment:
I'm not sure if this practice should be fixable because if we add the section in the readme automatically, the user is not pushed to edit the readme because we'll say, that he is practicing this practice, which wouldn't be true after all if there will be just the titles of sections.
Maybe if we will check that in every section there is some text. But I don't know if it's a good practice.

Reply:
oh, right we should allow the user to edit these themselves. abt the checking in sections :thinking: that the user can put text in a various format so will it be possible to cover them properly because having text will also not tell us abt whether its for the heading or something else

Reply:
It's quite tricky. Maybe we shouldn't fix this practice automatically at all. What do you think? Or the check can be like `is there any characters between the sections?

Reply:
Yeah fixing can be tricky and for the check part i guess we can't be sure as to what the user has entered between the sections so the practice eventually expects user to write the sections out properly.

Reply:
Yes, I think it would be better not to fix it automatically after all.